packages/snap-preact/components/src/components/Organisms/Layout/Layout.tsx

```diff
@@ -0,0 +1,432 @@
+			// default props
+			controller,
+			className: 'ss__layout__button-clearButton',
+			name: 'close-filters',
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+			treePath,
+		},
+		ApplyButton: {
+			// default props
+			controller,
+			className: 'ss__layout__button-applyButton',
+			name: 'apply-filters',
+
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+			treePath,
+		},
+	};
+
+	const defaultLang = {
+		titleText: {
+			value: titleText || props.lang?.titleText?.value,
+		},
+	};
+
+	//deep merge with props.lang
+	const lang = deepmerge(defaultLang, props.lang || {});
+	const mergedLang = useLang(lang as any, {
+		controller: controller,
+	});
+
+	function renderModule(module: ModuleNames) {
+		switch (module) {
+			case 'MobileSidebar':
+				if (controller.store.pagination.totalResults > 0) {
+					return <MobileSidebar controller={controller} {...subProps.MobileSidebar} />;
+				}
+				break;
+
+			case 'SearchHeader':
+				return <SearchHeader {...subProps.SearchHeader} />;
+
+			case 'FilterSummary':
+				return <FilterSummary {...subProps.FilterSummary} />;
+
+			case 'LayoutSelector':
+				if (themeStore && props.theme?.layoutOptions && props.theme.layoutOptions.length > 0) {
+					return (
+						<LayoutSelector
+							onSelect={(e, option) => {
+								if (option) {
+									themeStore?.layout.select(option);
+								}
+							}}
+							selected={themeStore?.layout.selected}
+							options={props.theme?.layoutOptions}
+							{...subProps.LayoutSelector}
+						/>
+					);
+				}
+				break;
+
+			case 'PaginationInfo':
+				return <PaginationInfo {...subProps.PaginationInfo} />;
+
+			case 'SortBy':
+				return <SortBy {...subProps.SortBy} />;
+
+			case 'PerPage':
+				return <PerPage {...subProps.PerPage} />;
+
+			case 'Button.toggleSideBar':
+				return (
+					toggleSideBarButton && (
+						<div className="ss__layout__button-toggleSideBarButtonWrapper">
+							<Button {...subProps.ToggleSideBarButton}>{cloneWithProps(toggleSideBarButton)}</Button>
+						</div>
+					)
+				);
+
+			case 'Pagination':
+				if (controller.config.settings?.infinite) {
+					return <LoadMore {...subProps.LoadMore} />;
+				} else {
+					return <Pagination {...subProps.Pagination} />;
+				}
+
+			case '_':
+				return <div className={`ss__layout__separator ss__layout__separator--${separatorIndex++}`} />;
+
+			case 'Banner.banner':
+				return <Banner {...subProps.Banner} type={ContentType.BANNER} name={'banner'} />;
+
+			case 'Banner.footer':
+				return <Banner {...subProps.Banner} type={ContentType.FOOTER} name={'footer'} />;
+
+			case 'Banner.header':
+				return <Banner {...subProps.Banner} type={ContentType.HEADER} name={'header'} />;
+
+			case 'Banner.left':
+				return <Banner {...subProps.Banner} type={ContentType.LEFT} name={'left'} />;
+
+			case 'Title':
```

Comment:
Should this really be here? Seems like we should have a legit component instead of this pseudo one... I would like to remove as many of the components as possible from here that are not actual components (if they don't exist in Storybook).